pandas/core/frame.py

```diff
@@ -6438,6 +6440,11 @@ def combine_first(self, other: DataFrame) -> DataFrame:
         other : DataFrame
             Provided DataFrame to use to fill null values.
 
+        preserve_dtypes : bool, default False
```

Comment:
we do not want to add a flag for this. simply change it.

Please add some examples for this behaivor

Reply:
I was thinking that maybe it is a good idea to keep the current behavior as default, and provide the new behavior as an option

Reply:
no its better to just fix this, you can add a whatsnew note in 1.3. everywhere else we cast to common dtypes, this should be no different.

Reply:
I'm running into some failed tests that exceed my understand of the lib. Is it expected that if a Series is constructed from a list of ``None`` then the result of this combined with some other Series should have the latter's dtype (coercing to the respective ``NaN`` value)?

Reply:
list of None -> object, so combined -> object